internal/service/domain/sensor/sensor.go

```diff
@@ -21,11 +23,15 @@ func NewSensorService(
 	treeRepo storage.TreeRepository,
 	flowerbedRepo storage.FlowerbedRepository,
 ) service.SensorService {
-	return &SensorService{
+	s := &SensorService{
 		sensorRepo:    sensorRepo,
 		treeRepo:      treeRepo,
 		flowerbedRepo: flowerbedRepo,
 	}
+
+	go s.RunStatusUpdater(context.Background(), 1*time.Hour) // Run every 1 hour
```

Comment:
I wouldn't run it here because it runs every time you create the service. I would put it in `server.go`, there is also the plugin cleanup service.

Reply:
You can also use the context from the application, rather than from the background.